### PR TITLE
Update Java SDK v8 deprecation removals

### PR DESCRIPTION
**Pull request by adinauer**
We've removed some APIs in Java SDK v8. This PR replaces / removes docs for those.

Here's the changelog: https://github.com/getsentry/sentry-java/blob/8.x.x/CHANGELOG.md

## DESCRIBE YOUR PR
*Tell us what you're changing and why. If your PR **resolves an issue**, please link it so it closes automatically.*

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [ ] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## LEGAL BOILERPLATE

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Review by lizokm — APPROVED** [review]
Thanks for updating!

File: docs/platforms/android/configuration/options.mdx [hunk 1]
```diff
@@ -258,7 +258,7 @@ Default: set to `android.content.Context.getCacheDir()/sentry`.
 
 </ConfigKey>
 
-<ConfigKey name="shutdown-timeout">
+<ConfigKey name="shutdown-timeout-millis">
 
 Controls how many seconds to wait before shutting down. Sentry SDKs send events from a background queue. This queue is given a certain amount to drain pending events. The default is SDK specific but typically around two seconds. Setting this value too low may cause problems for sending events from command line applications. Setting the value too high will cause the application to block for a long time for users experiencing network connectivity problems.
```
Comment (lbloder) [inline docs/platforms/android/configuration/options.mdx:263]: (m) I know this has been here before, but maybe we could remove the mention of "command line applications" as this is not relevant for android.

Maybe just:
```suggestion
Controls how many seconds to wait before shutting down. Sentry SDKs send events from a background queue. This queue is given a certain amount to drain pending events. The default is SDK specific but typically around two seconds. Setting this value too low may cause problems for sending events from the application. Setting the value too high will cause the application to block for a long time for users experiencing network connectivity problems.
```

**Review by lbloder — COMMENTED** [review]
Not sure if this should be part of this pr, however, compared to the migration guide, here are some things I found.

- `enableTracing` option (a.k.a `enable-tracing`) is still in the docs
- In `docs/platforms/android/metrics/index.mdx` we should mention, that metrics are no longer supported in 8.x.x or remove the metrics docs altogether

**Comment by adinauer** [comment]
> enableTracing option (a.k.a enable-tracing) is still in the docs
In docs/platforms/android/metrics/index.mdx we should mention, that metrics are no longer supported in 8.x.x or remove the metrics docs altogether

We have separate issues to track changing / removing those.

**Review by lbloder — APPROVED** [review]
LGTM 👍

**Comment by adinauer** [comment]
I've retargeted this PR to a new branch so we can collect all changes we want to merge after v8 GAs.

**Pull request merged**